backend-api/src/functions/common/logging/setupLogger.ts

```diff
@@ -1,8 +1,43 @@
 import { Context } from "aws-lambda";
 import { logger } from "./logger";
 
-export const setupLogger = (context: Context) => {
+export const setupLogger = (
+  context: Context,
+  userAgent: string | undefined,
+) => {
   logger.resetKeys();
   logger.addContext(context);
-  logger.appendKeys({ functionVersion: context.functionVersion });
+  logger.appendKeys({
+    functionVersion: context.functionVersion,
+    userAgent: buildUserAgent(userAgent),
+  });
+};
+
+export const buildUserAgent = (
+  userAgent: string | undefined,
+): object | undefined => {
+  if (userAgent === undefined || userAgent === "") {
```

Comment:
suggestion (non-blocking): as `userAgent` is `string` or `undefined`, you can simplify this check


```suggestion
  if (!userAgent) {
```